contrib/direnv

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
+
+# Look for the .config folder in the current directory and up
+find_prj_config() (
+	local old_pwd
+	while [[ $old_pwd != $PWD ]]; do
+		if [[ -d .config ]]; then
+			echo "$PWD"
+			return 0
+		fi
+		old_pwd=$PWD
+		cd ..
+	done
+	# We're at the top and didn't find anything
+	log_error "ERROR: could not find project root" >&2
+	return 1
+)
+
+{
+	: "${XDG_CONFIG_HOME:=${HOME}/.config}"
+	: "${XDG_CACHE_HOME:=${HOME}/.cache}"
+	: "${XDG_DATA_HOME:=${HOME}/.local/share}"
+
+	: "${GIT_DIR:=$(git rev-parse --absolute-git-dir 2>/dev/null)}"
+
+	# PRJ_ROOT
+	_prj_root="${GIT_DIR%/.git}"
+	: "${PRJ_ROOT:=${_prj_root:=$(find_prj_config)}}"
+
+	# PRJ_CONFIG_HOME - always local to the project
+	: "${PRJ_CONFIG_HOME:=${PRJ_ROOT}/.config}"
+	mkdir -p "${PRJ_CONFIG_HOME}"
+
+	if [[ -z "${PRJ_ID:-}" && -f "${PRJ_CONFIG_HOME}/prj_id" ]]; then
+		PRJ_ID=$(<"${PRJ_CONFIG_HOME}/prj_id")
+	fi
+
+	# PRJ_RUNTIME_DIR - always local to the project
+	: "${PRJ_RUNTIME_DIR:=${PRJ_ROOT}/.run}"
+	mkdir -p "${PRJ_RUNTIME_DIR}"
+
+	# PRJ_CACHE_HOME - shared if PRJ_ID is set
+	if [[ -z "${PRJ_CACHE_HOME:-}" ]]; then
+		if [[ -n "${PRJ_ID:-}" ]]; then
+			PRJ_CACHE_HOME="${XDG_CACHE_HOME}/prj/${PRJ_ID}"
+		else
+			PRJ_CACHE_HOME="${PRJ_ROOT}/.cache"
+		fi
+	fi
+	mkdir -p "${PRJ_CACHE_HOME}"
+
+	# PRJ_DATA_HOME - shared if PRJ_ID is set
+	if [[ -z "${PRJ_DATA_HOME:-}" ]]; then
+		if [[ -n "${PRJ_ID:-}" ]]; then
+			PRJ_DATA_HOME="${XDG_DATA_HOME}/prj/${PRJ_ID}"
+		else
+			PRJ_DATA_HOME="${PRJ_ROOT}/.data"
+		fi
+	fi
+	mkdir -p "${PRJ_DATA_HOME}"
+	PRJ_STATE_HOME="$PRJ_DATA_HOME"
+
+	# PRJ_PATH - shared if PRJ_ID is set
+	if [[ -z "${PRJ_PATH:-}" ]]; then
+		if [[ -n "${PRJ_ID:-}" ]]; then
+			PRJ_PATH="${HOME}/.local/bin/prj/${PRJ_ID}"
+		else
+			PRJ_PATH="${PRJ_ROOT}/.bin"
+		fi
+	fi
+	mkdir -p "${PRJ_PATH}"
+
+	# Git ignore all `PRJ_*` in the main checkout
+	# i.e. if we're not in a worktree, that is: GIT_DIR is not a file
```

Comment:
I think this comment is perhaps incorrect. 

`GIT_DIR` as set by `git rev-parse --absolute-git-dir` **could** be set, but we could be outside a worktree. For example, if we were in a "bare" git repository or within the `.git` git subdir within the main worktree then `GIT_DIR` would still be set.

To be fair, for the inverse, i.e. if it's **not* set, then we're definitely **not in** work tree. 

Putting aside the reservations below about doing this at all, if you were to keep it, I'd advise something like;

 ```suggestion
	# Git ignore all `PRJ_*` in the worktree for this repository.
	# Skip this if we're not in a git repository
	# i.e. if we're not in a git repository, GIT_DIR is not a file
```

To get some of this straight in my head for this comment and others, I wrote this [little hacky demo script](https://gist.github.com/cscutcher/e55c6351c7dd60ef7743186ad89eccb8#file-demo-sh) with output that I hope demonstrates;

```console
$ cd example_repo && echo /tmp
/tmp



$ # Example of different git values within main worktree...
$ git worktree add ../example_repo_worktree
Preparing worktree (checking out 'example_repo_worktree')
HEAD is now at 03370b6 Initial commit
$ git rev-parse --show-toplevel
/tmp/example_repo
$ git rev-parse --absolute-git-dir
/tmp/example_repo/.git
$ git rev-parse --is-inside-work-tree
true
$ git rev-parse --is-inside-git-dir
false



$ cd .git
$ git rev-parse --show-toplevel
fatal: this operation must be run in a work tree
$ git rev-parse --absolute-git-dir
/tmp/example_repo/.git
$ git rev-parse --is-inside-work-tree
false
$ git rev-parse --is-inside-git-dir
true



$ # Example of different git values within a linked worktree...
$ cd ../../example_repo_worktree
$ git rev-parse --show-toplevel
/tmp/example_repo_worktree
$ git rev-parse --absolute-git-dir
/tmp/example_repo/.git/worktrees/example_repo_worktree
$ git rev-parse --is-inside-work-tree
true
$ git rev-parse --is-inside-git-dir
false



$ # Just for completeness, the output of these when outside a git repo
$ cd /tmp
$ git rev-parse --show-toplevel
fatal: not a git repository (or any of the parent directories): .git
$ git rev-parse --absolute-git-dir
fatal: not a git repository (or any of the parent directories): .git
$ git rev-parse --is-inside-work-tree
fatal: not a git repository (or any of the parent directories): .git
$ git rev-parse --is-inside-git-dir
fatal: not a git repository (or any of the parent directories): .git
```

Reply:
fixed in https://github.com/numtide/prj-spec/pull/4/commits/3ea5e756ca4c228aeb8586eb13fa899e295fdca6